packages/libs/src/sdk/api/tracks/TracksApi.ts

```diff
@@ -366,4 +377,173 @@ export class TracksApi extends GeneratedTracksApi {
       ...advancedOptions
     })
   }
+
+  /**
+   * Purchases stream or download access to a track
+   *
+   * @hidden
+   */
+  public async purchase(params: PurchaseTrackRequest) {
+    const {
+      userId,
+      trackId,
+      extraAmount: extraAmountNumber = 0,
+      walletAdapter
+    } = await parseParams('purchase', PurchaseTrackSchema)(params)
+
+    const contentType = 'track'
+    const mint = 'USDC'
+
+    // Fetch track
+    this.logger.debug('Fetching track...', { trackId })
+    const { data: track } = await this.getTrack({
+      trackId: params.trackId // use hashed trackId
+    })
+
+    // Validate purchase attempt
+    if (!track) {
+      throw new Error('Track not found.')
+    }
+
+    if (!track.isStreamGated && !track.isDownloadGated) {
+      throw new Error('Attempted to purchase free track.')
+    }
+
+    if (track.user.id === params.userId) {
+      throw new Error('Attempted to purchase own track.')
+    }
+
+    let numberSplits: UsdcGate['splits'] = {}
+    let centPrice: number
+    let accessType: 'stream' | 'download' = 'stream'
+
+    // Get conditions
+    if (track.streamConditions && 'usdcPurchase' in track.streamConditions) {
+      centPrice = track.streamConditions.usdcPurchase.price
+      numberSplits = track.streamConditions.usdcPurchase.splits
+    } else if (
+      track.downloadConditions &&
+      'usdcPurchase' in track.downloadConditions
+    ) {
+      centPrice = track.downloadConditions.usdcPurchase.price
+      numberSplits = track.downloadConditions.usdcPurchase.splits
+      accessType = 'download'
+    } else {
+      throw new Error('Track is not available for purchase.')
+    }
+
+    // Check if already purchased
+    if (
+      (accessType === 'download' && track.access?.download) ||
+      (accessType === 'stream' && track.access?.stream)
+    ) {
+      throw new Error('Track already purchased')
+    }
+
+    let extraAmount = USDC(extraAmountNumber).value
+    const total = USDC(centPrice / 100.0).value + extraAmount
+    this.logger.debug('Purchase total:', total)
+
+    // Convert splits to big int and spread extra amount to every split
+    const splits = Object.entries(numberSplits).reduce(
```

Comment:
Is there an easy way for us to test this API? There are a few spots of logic in here that feel like very explicit behavior that we wouldn't want to accidentally regress.